### PR TITLE
PAPP-23589 - Add md5 support for non-fips phantom instances

### PR DESCRIPTION
**Pull request by chunmanjimmyf**
…instances

Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [X] Testing of all the changes has been performed (for bug fixes / features)
- [X] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [X] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [X] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [X] Verify all checks are passing.
- [X] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [X] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Currently we generate SDIs in SHA256 format when creating artifacts on ingestion.

## What is the new behavior? (OPTIONAL)
- For non-fips Phantom instances, we will generate SDIs in MD5 format so we do not create duplicate artifacts if the artifact already existed prior.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: splunk_connector.py [hunk 1]
```diff
@@ -670,14 +690,28 @@ def _get_host_events(self, param):
...
+        ret_val, last_n_days = self._validate_integer(action_result, param.get(consts.SPLUNK_JSON_LAST_N_DAYS),
+            consts.SPLUNK_LAST_N_DAYS_KEY)
         if phantom.is_fail(ret_val):
             return action_result.get_status()
 
         search_query = 'search host="{0}"{1}'.format(ip_hostname, ' earliest=-{0}d'.format(last_n_days) if last_n_days else '')
 
         return self._run_query(search_query, action_result)
 
+    def _get_fips_enabled(self):
```
Comment (chunmanjimmyf) [inline splunk_connector.py:702]: FIPS change.

File: splunk_connector.py [hunk 2]
```diff
@@ -765,9 +799,18 @@ def _on_poll(self, param):
             if self._python_version == 3:
                 input_str = UnicodeDammit(input_str).unicode_markup.encode('utf-8')
 
-            sha256.update(input_str)
```
Comment (chunmanjimmyf) [inline splunk_connector.py:768]: FIPS change.

**Comment by chunmanjimmyf** [comment]
Integration test is passing, but one test is timing out.

File: splunk_connector.py [hunk 3]
```diff
@@ -670,14 +680,28 @@ def _get_host_events(self, param):
...
         return self._run_query(search_query, action_result)
 
+    def _get_fips_enabled(self):
+        try:
+            from phantom_common.install_info import is_fips_enabled
+        except ImportError:
+            return False
+
+        fips_enabled = is_fips_enabled()
+        if (fips_enabled):
```
Comment (alexa-phantom) [inline splunk_connector.py:699]: Don't need parenthesis

**Review by alexa-phantom — CHANGES_REQUESTED** [review]

File: splunk_connector.py [hunk 4]
```diff
@@ -394,8 +390,11 @@ def _check_for_es(self, action_result):
     def _resolve_event_id(self, sidandrid, action_result, kwargs_create=dict()):
         """Query the splunk instance using the SID+RID of the notable to find the notable ID"""
 
-        search_query = r'search [| makeresults | eval myfield = "{}" | rex field=myfield "^(?<sid>.*)\+(?<rid>\d*(\.\d+)?)"'.format(sidandrid)
-        search_query += r' | eval search = "( (sid::" . sid . " OR orig_sid::" . sid . ") (rid::" . rid . " OR orig_rid::" . rid . ") )" | table search] `notable` | table event_id'
+        search_query = r'search [| makeresults | eval myfield = "{}"'.format(sidandrid)
+        search_query += r' | rex field=myfield "^(?<sid>.*)\+(?<rid>\d*(\.\d+)?)"'
+        search_query += r' | eval search = "( (sid::" . sid . " OR orig_sid::" . sid . ")'
+        search_query += r' (rid::" . rid . " OR orig_rid::" . rid . ") )"'
+        search_query += r' | table search] `notable` | table event_id'
```
Comment (rgil-splunk) [inline splunk_connector.py:397]: can we make put this in the constants file?

File: splunk_connector.py (hunk 3 above)
Comment (rgil-splunk) [inline splunk_connector.py:699]:
```suggestion
        if fips_enabled:
```

File: splunk_connector.py [hunk 5]
```diff
@@ -670,14 +680,28 @@ def _get_host_events(self, param):
...
 
+    def _get_fips_enabled(self):
+        try:
+            from phantom_common.install_info import is_fips_enabled
+        except ImportError:
+            return False
+
+        fips_enabled = is_fips_enabled()
+        if (fips_enabled):
+            self.debug_print('fips is enabled')
```
Comment (rgil-splunk) [inline splunk_connector.py:700]:
```suggestion
            self.debug_print('FIPS is enabled')
```

File: splunk_connector.py [hunk 6]
```diff
@@ -670,14 +680,28 @@ def _get_host_events(self, param):
...
+        else:
+            self.debug_print('fips is not enabled')
```
Comment (rgil-splunk) [inline splunk_connector.py:702]:
```suggestion
            self.debug_print('FIPS is not enabled')
```

File: splunk_connector.py [hunk 7]
```diff
@@ -765,9 +789,18 @@ def _on_poll(self, param):
             if self._python_version == 3:
                 input_str = UnicodeDammit(input_str).unicode_markup.encode('utf-8')
 
-            sha256.update(input_str)
+            fips_enabled = self._get_fips_enabled()
+            # if fips is not enabled, we should continue with our existing md5 usage for generating SDIs
+            # to not impact existing customers
+            if (not fips_enabled):
```
Comment (rgil-splunk) [inline splunk_connector.py:795]:
```suggestion
            if not fips_enabled:
```

File: splunk_connector.py [hunk 8]
```diff
@@ -974,7 +1008,8 @@ def _list_alerts(self, param, action_result=None):
             self.debug_print("ss_names", ss_names)
             ss_query = 'ss_name = {}'.format(' OR ss_name = '.join(ss_names))
 
-        query = 'search index=_audit action=alert_fired {0} | head {1} | fields ss_name sid trigger_time severity'.format(ss_query, count)
+        query = "search index=_audit action=alert_fired {0} | head {1} | \
+            fields ss_name sid trigger_time severity".format(ss_query, count)
```
Comment (rgil-splunk) [inline splunk_connector.py:1012]: ditto about putting this in the constants file

**Review by rgil-splunk — APPROVED** [review]

**Comment by rgil-splunk** [comment]
looks like compilation is failing (see static tests)

**Comment by chunmanjimmyf** [comment]
Overall QA result from integration tests are showing pass. However the tests are taking longer than 1 hour which is the time out and causing the integration test to fail.

File: splunk_connector.py [hunk 9]
```diff
@@ -141,7 +131,8 @@ def initialize(self):
             self._state = self.load_state()
             if self._state is None:
                 self.debug_print("Please check the owner, owner group, and the permissions of the state file")
-                self.debug_print("The phantom user should be having correct access rights and ownership for the corresponding state file (refer readme file for more information)")
+                self.debug_print("The phantom user should be having correct access rights and ownership for the \
```
Comment (alexa-phantom) [inline splunk_connector.py:134]:
```suggestion
                self.debug_print("The phantom user should have correct access rights and ownership for the \
```

**Review by alexa-phantom — APPROVED** [review]

**Pull request merged**